### PR TITLE
Add IFEM-ShellEx and IFEM-BeamGeometry to common app build

### PR DESCRIPTION
**Pull request by kmokstad**
Such that jenkins can pick up these modules too.

### CODE REVIEW

File: Apps/CMakeLists.txt [hunk 1]
```diff
@@ -49,9 +49,26 @@ else()
 endif()
 if(EXISTS ${PROJECT_SOURCE_DIR}/${ELASTICITY_SRC})
 
+  message(STATUS "Including ${PROJECT_SOURCE_DIR}/${ELASTICITY_SRC}")
```
Comment (akva2) [inline Apps/CMakeLists.txt:52]: this is useful. can you do this for every module while at it?

**Review by akva2 — APPROVED** [review]

**Comment by kmokstad** [comment]
jenkins build this with downstreams ifem-elasticity=160 ifem-beamex=38 please

**Comment by akva2** [comment]
you probably want to update jenkins/build.sh in the repos that get a new upstream.

File: Apps/CMakeLists.txt (hunk 1 above)
Comment (kmokstad) [reply Apps/CMakeLists.txt:49]: Done, by adding a macro to avoid too many repetitive code block.

**Comment by kmokstad** [comment]
jenkins build this with downstreams ifem-elasticity=160 ifem-beamex=38 please

**Comment by kmokstad** [comment]
jenkins build this with downstreams ifem-elasticity=160 ifem-beamex=38 please

**Comment by kmokstad** [comment]
jenkins build this with downstreams ifem-elasticity=160 ifem-beamex=38 please

**Comment by kmokstad** [comment]
jenkins build this with downstreams ifem-elasticity=160 ifem-beamex=38 please

**Comment by kmokstad** [comment]
I can't figure out the build failure here. Building the new BeamGeometry module, which reside in a separate repository requires that header files from IFEM-Elasticity are found. How can they be made available? Thought it should be handled by the jenkins/build.sh file of this repo, by declaring IFEM-Elasticity as upstream, but did not seem to help although similar construct for IFEM-BeamEx. The logic behind the up/down/side-stream handling is rather complicated, so @akva2 I'll need your assistance on this.

**Comment by kmokstad** [comment]
jenkins build this with downstreams ifem-elasticity=160 ifem-beamex=38 please

**Comment by kmokstad** [comment]
jenkins build this with downstreams ifem-elasticity=160 ifem-beamex=38 please

**Comment by kmokstad** [comment]
jenkins build this with downstreams ifem-elasticity=160 ifem-beamex=38 please

**Comment by kmokstad** [comment]
jenkins build this with downstreams ifem-elasticity=160 ifem-beamex=38 please

**Comment by kmokstad** [comment]
jenkins build this with downstreams ifem-elasticity=160 ifem-beamex=38 please

**Comment by kmokstad** [comment]
jenkins build this with downstreams ifem-elasticity=160 ifem-beamex=38 please

**Comment by kmokstad** [comment]
jenkins build this with downstreams ifem-elasticity=160 ifem-beamex=38 please

**Comment by kmokstad** [comment]
jenkins build this with downstreams ifem-elasticity=160 ifem-beamex=38 please

**Comment by kmokstad** [comment]
jenkins build this with downstreams ifem-elasticity=160 ifem-beamex=38 please

**Comment by kmokstad** [comment]
This finally passes now. The clue was to check whether we are in a top-level project (by comparing PROJECT_NAME and CMAKE_PROJECT_NAME) before adding the dummy testapps target in IFEM-BeamGeometry. The same thing should be possible using the IFEM_COMMON_APP_BUILD variable, but that did not work for some reason. The inter-dependency between IFEM-BeamGeometry and IFEM-Elasticity is awkward, I know, but as long the central build does not fail, I'm happy.

**Pull request merged**